server/storage/wal/wal_test.go

```diff
@@ -639,7 +639,9 @@ func TestOpenWithMaxIndex(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	defer w.Close()
+	defer func() {
+		w.Close()
+	}()
```

Comment:
I would suggest to move the `defer w.Close()` to the place right after the `Open` (about line 653) for better readability.